proxy_presentation.php

```diff
@@ -60,14 +60,37 @@
     }
 
     $bbbinstanceid = $meetingidparts[2];
+    $basemeetingid = $meetingidparts[0];
 
     // MeetingID can sometimes have [xxx] at the end. Needs sanitisation.
     if (strpos($bbbinstanceid, '[') !== false) {
         $bbbinstanceid = substr($bbbinstanceid, 0, strpos($bbbinstanceid, '['));
     }
 
-    $viewinstance = bigbluebuttonbn_view_instance_bigbluebuttonbn($bbbinstanceid);
-    require_login($viewinstance['course'], true, $viewinstance['cm']);
+
+    // Check if the user has appropriate access to the meeting resources.
+    $meetingsinstances = $DB->get_records('bigbluebuttonbn', ['meetingid' => $basemeetingid], '', 'id, course');
+    $access = false;
+    $counter = 1;
+    $maxcount = count($meetingsinstances);
+    foreach ($meetingsinstances as $bbbinstance) {
+        try {
+            $viewinstance = bigbluebuttonbn_view_instance_bigbluebuttonbn($bbbinstance->id);
+            // NOTE: Bit dirty, but felt better than pulling out specific internals.
+            // And does a normal redirect on the last check (e.g. if it fails).
+            require_login($viewinstance['course'], true, $viewinstance['cm'], $setwantsurltome = true, $preventredirect = ($counter !== $maxcount));
```

Comment:
I think we should use has_capability here instead, and just have a boolean check, and we can just break on the first true response, else exception

Reply:
and then require_login after the break out of the loop once we find a matching activity

Reply:
Done, except kept the loop, only attempts require_login on initial pass and breaks loops on exceptions or if the require_login passes

Reply:
Removed loop, was less lines and clearer intent without. No need to explicitly throw exception, require_login does that for us. On capability checks, as-is there will be more DB lookups if require_login checks was performed as is. Did you want me to explicitly throw an error/exception regarding access if the user does not have capabilities for any of the returning activities?